src/timecode.rs

```diff
@@ -497,17 +498,122 @@ impl Timecode {
     ```rust
     # use vtc::{Timecode, rates};
     let tc = Timecode::with_frames("01:00:00:00", rates::F23_98).unwrap();
-    assert_eq!("5400+00", tc.feet_and_frames())
+    assert_eq!("5400+00", tc.feet_and_frames_35mm_4p())
     ```
     */
-    pub fn feet_and_frames(&self) -> String {
-        let result = div_mod_floor(abs(self.frames()), FRAMES_PER_FOOT);
-        let feet = result.0;
-        let frames = result.1;
+    pub fn feet_and_frames_35mm_4p(&self) -> String {
+        self.feet_and_frames_impl(4, PERFS_PER_FOOT_35)
+    }
+
+    /**
+     Returns the feet and frames representation in 16mm footage.
+
+     # What it is
+
+     On 16mm film, there are forty frames of film in each foot, one perforation
+     per frame. However, 16mm film is edge coded every six inches, with twenty
+     frames per code, so the footage "1+19" is succeeded by "2+0".
+
+     # Where you see it
+
+     16mm telecines, 16mm edge codes.
+
+     # Examples
+
+    ```rust
+    # use vtc::{Timecode, rates};
+    let tc = Timecode::with_frames("00:00:01:00", rates::F23_98).unwrap();
+    assert_eq!("1+04", tc.feet_and_frames_16mm())
+    ```
+     */
+    pub fn feet_and_frames_16mm(&self) -> String {
+        self.feet_and_frames_impl(1, PERFS_PER_6INCHES_16)
+    }
+
+    /**
+    Returns the feet and frames representation in 35mm, 2-perf footage.
+
+    # What it is
+
+    35mm 2-perf film records 32 frames in a foot of film, instead of the usual 16.
+    This creates a negative image with a wide aspect ratio using standard spherical
+    lenses and consumes half the footage per minute running time as standard 35mm,
+    while having a grain profile somewhat better than 16mm while not as good as
+    standard 35mm.
+
+    # Where you see it
+
+    35mm 2-perf formats are uncommon though still find occasional use, the process is
```

Comment:
Thanks for adding all this. It's really good background!

Reply:
Done